content/blog/2024-05-23-linebender-roadmap.md

```diff
@@ -0,0 +1,248 @@
+
+- **Priority:** High.
+- **Assigned:** Olivier Faure.
+- **Description:** Right now widgets can use context methods in their different passes to request that other passes be ran. What passes can run what other passes isn't clearly documented, especially around edge cases. We should write an RFC to nail down the passes, how they can communicate, and the rationale around them.
+- **PR:** TBD.
+
+### Rewrite documentation
+
+- **Priority:** Medium.
+- **Assigned:** Olivier Faure.
+- **Description:** Masonry's documentation hasn't been updated since the port to Winit. Once other major changes are implemented, the documentation should be brought up to date. We should avoid procrastinating on documentation too much just because the architecture is still in progress.
+- **Issue:** [xilem#389](https://github.com/linebender/xilem/issues/389)
+
+### Improve focus handling
+
+- **Priority:** Medium.
+- **Assigned:** Olivier Faure.
+- **Description:** We should improve tab focus in Masonry. Right now tab focus has some bugs, doesn't cover buttons, isn't updated by clicks, and uses architecture inherited from Druid that could be strongly simplified once other architectural changes are completed.
+- **Issue:** [xilem#390](https://github.com/linebender/xilem/issues/390)
+
+### Improve pointer status handling
+
+- **Priority:** Medium.
+- **Assigned:** Olivier Faure.
+- **Description:** We should centralize and improve how Masonry handles pointers. In particular, we should add an explicit "pointer capture" API that makes other widgets not get events from that pointer until the pointer is released or lost.
+- **Issue:** [xilem#312](https://github.com/linebender/xilem/issues/312)
+
+
+## Parley
+
+### Create abstractions for text selection and editing
+
+- **Priority:** Highest.
+- **Assigned:** Chad Brokaw, Aaron Muir Hamilton.
+- **Description:** We want Parley to implement types and methods that Masonry and other editors will be able to use to create a text-editing widget, or to handle selection in non-editable text. The initial implementation may be located in Masonry.
+- **Issue:** [parley#52](https://github.com/linebender/parley/issues/52)
+
+### Inline block layout
+
+- **Priority:** High.
+- **Assigned:** Chad Brokaw.
+- **Description:** Parley should support laying out inline content the way web browsers do, where blocks are splices into your text and should be laid out according to available inline space the same way words are.
+- **Issue:** [parley#25](https://github.com/linebender/parley/issues/25)
+- **PRs:** [parley#67](https://github.com/linebender/parley/pull/67), [parley#76](https://github.com/linebender/parley/pull/76)
+
+### Write adequate documentation
+
+- **Priority:** High.
+- **Assigned:** Nico Burns.
+- **Description:** Parley currently has virtually no documentation. If we want it to become a fundational crate, we need to explain to people how to use it.
+- **PR:** [parley#26](https://github.com/linebender/parley/pull/26)
```

Comment:
We really should try and just merge this "as-is"; it's not doing any good just sat as a draft.